src/cli/serve/deprecated_config.js

```diff
@@ -0,0 +1,62 @@
+import { forOwn, has, transform } from 'lodash';
+import { red } from 'ansicolors';
+import { memoize } from 'lodash';
+
+const logWarning = memoize(function (key, message) {
+  console.log(red('WARNING:'), message);
+});
+
+const legacySettings = {
+  // server
+  port: 'server.port',
+  host: 'server.host',
+  pid_file: 'pid.file',
+  ssl_cert_file: 'server.ssl.cert',
+  ssl_key_file: 'server.ssl.key',
+
+  // logging
+  log_file: 'logging.dest',
+
+  // kibana
+  kibana_index: 'kibana.index',
+  default_app_id: 'kibana.defaultAppId',
+
+  // es
+  ca: 'elasticsearch.ssl.ca',
+  elasticsearch_preserve_host: 'elasticsearch.preserveHost',
+  elasticsearch_url: 'elasticsearch.url',
+  kibana_elasticsearch_client_crt: 'elasticsearch.ssl.cert',
+  kibana_elasticsearch_client_key: 'elasticsearch.ssl.key',
+  kibana_elasticsearch_password: 'elasticsearch.password',
+  kibana_elasticsearch_username: 'elasticsearch.username',
+  ping_timeout: 'elasticsearch.pingTimeout',
+  request_timeout: 'elasticsearch.requestTimeout',
+  shard_timeout: 'elasticsearch.shardTimeout',
+  startup_timeout: 'elasticsearch.startupTimeout',
+  verify_ssl: 'elasticsearch.ssl.verify',
+};
+
+const deprecatedSettings = {
+  'server.xsrf.token': 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.'
+};
+
+// transform legeacy options into new namespaced versions
+export function rewriteDeprecatedConfig(object) {
```

Comment:
![image](https://cloud.githubusercontent.com/assets/1329312/14501744/cd64fefc-015c-11e6-8d16-539bd928b32a.png)

(I'd love to know how many times I've needed you to tell me to write more tests. Sorry man)